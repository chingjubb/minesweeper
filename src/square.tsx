import { Tile } from './minesweeper_reducer';
import styles from './square.module.css';
import classnames from 'classnames';

type SquareProps = {
	onClick: () => void;
	onRightClick: () => void;
} & Tile;

export const BOMB_ICON = 'ðŸ’£';
export const FLAG_ICON = 'ðŸš©';

export const Square = (props: SquareProps) => {
	const { isBomb, flagged, clicked, onClick, value, onRightClick } = props;
	if (flagged) {
		return <div onContextMenu={(e) => {
						e.preventDefault();
						onRightClick();
					}}
					onClick={onClick}
					className={classnames(styles.tile, styles.flagged)}>{FLAG_ICON}</div>
	}

	if (!clicked) {
		return <div onClick={onClick}
					onContextMenu={(e) => {
						e.preventDefault();
						onRightClick();
					}}
					className={classnames(styles.tile,
										  styles.unclickedTile)} />
	}
	
	if (isBomb) {
		return <div className={classnames(styles.bomb,
										  styles.tile)}>
					{BOMB_ICON}
				</div>
	}
	const getValueStyle = (value: number) => {
		switch (value) {
			case 1: return styles.oneBomb;
			case 2: return styles.twoBombs;
			case 3: return styles.threeBombs;
			case 4: return styles.fourBombs;
			case 5: return styles.fiveBombs;
			case 6: return styles.sixBombs;
			case 7: return styles.sevenBombs;
			case 8: return styles.eightBombs;
			default:
				return '';
		}
	}
	return <div className={classnames(styles.clickedTile,
								      styles.tile, getValueStyle(value))}>
				{value > 0 ? value : ''}
			</div>
};
