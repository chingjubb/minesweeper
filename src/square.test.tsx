import renderer from 'react-test-renderer';
import { Square } from './square';

test('unclicked tile', () => {
 	const component = renderer.create(
		<Square onRightClick={()=> {}} onClick={()=>{}} flagged={false} value={1} isBomb row={0} column={0} clicked={false} />,
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('flagged tile', () => {
 	const component = renderer.create(
		<Square onRightClick={()=> {}} onClick={()=>{}} flagged value={1} isBomb row={0} column={0} clicked={false} />,
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Bomb tile', () => {
 	const component = renderer.create(
		<Square onRightClick={()=> {}} onClick={()=>{}} flagged={false} value={1} isBomb row={0} column={0} clicked />,
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

describe('clicked tile ', ()=>{
	for (let i = 0; i <= 8; i++) {
		test(`clicked tile value = ${i}`, () => {
		 	const component = renderer.create(
				<Square onRightClick={()=> {}} onClick={()=>{}} flagged={false} value={i} isBomb={false} row={0} column={0} clicked />,
			);
			let tree = component.toJSON();
			expect(tree).toMatchSnapshot();
		});
	}
});
