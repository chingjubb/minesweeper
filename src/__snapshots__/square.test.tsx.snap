// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bomb tile 1`] = `
<div
  className="bomb tile"
>
  ðŸ’£
</div>
`;

exports[`clicked tile  clicked tile value = 0 1`] = `
<div
  className="clickedTile tile"
>
  
</div>
`;

exports[`clicked tile  clicked tile value = 1 1`] = `
<div
  className="clickedTile tile oneBomb"
>
  1
</div>
`;

exports[`clicked tile  clicked tile value = 2 1`] = `
<div
  className="clickedTile tile twoBombs"
>
  2
</div>
`;

exports[`clicked tile  clicked tile value = 3 1`] = `
<div
  className="clickedTile tile threeBombs"
>
  3
</div>
`;

exports[`clicked tile  clicked tile value = 4 1`] = `
<div
  className="clickedTile tile fourBombs"
>
  4
</div>
`;

exports[`clicked tile  clicked tile value = 5 1`] = `
<div
  className="clickedTile tile fiveBombs"
>
  5
</div>
`;

exports[`clicked tile  clicked tile value = 6 1`] = `
<div
  className="clickedTile tile sixBombs"
>
  6
</div>
`;

exports[`clicked tile  clicked tile value = 7 1`] = `
<div
  className="clickedTile tile sevenBombs"
>
  7
</div>
`;

exports[`clicked tile  clicked tile value = 8 1`] = `
<div
  className="clickedTile tile eightBombs"
>
  8
</div>
`;

exports[`flagged tile 1`] = `
<div
  className="tile flagged"
  onClick={[Function]}
  onContextMenu={[Function]}
>
  ðŸš©
</div>
`;

exports[`unclicked tile 1`] = `
<div
  className="tile unclickedTile"
  onClick={[Function]}
  onContextMenu={[Function]}
/>
`;
